### PR TITLE
hosted v3: init

### PR DESCRIPTION
**Pull request by mPaella**
## Description

init structure + some logic for hosted v3

## Test plan

local

https://github.com/user-attachments/assets/e5922587-c354-4c0b-a822-1fdac152c693

### CODE REVIEW

File: packages/client/ui/react-ui/src/components/hosted/v3/CrossmintHostedCheckoutV3.tsx [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+        hostedCheckoutService.createWindow();
+
+        if (onClick) {
+            onClick(e);
+        }
+    }
+
+    return (
+        <button onClick={_onClick} {...restButtonProps}>
+            Pay with Crossmint
```
Comment (mPaella) [inline packages/client/ui/react-ui/src/components/hosted/v3/CrossmintHostedCheckoutV3.tsx:41]: just temp for now, no styles on the button yet either

File: packages/client/base/src/types/hosted/v3/CrossmintHostedCheckoutV3Props.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
+import type { Locale } from "@/types";
+import type { EmbeddedCheckoutV3LineItem, EmbeddedCheckoutV3Payment, EmbeddedCheckoutV3Recipient } from "@/types/embed";
+
+export interface CrossmintHostedCheckoutV3Props {
```
Comment (pgarciaegido) [inline packages/client/base/src/types/hosted/v3/CrossmintHostedCheckoutV3Props.ts:4]: Have this interface been reviewed?

Comment (mPaella) [reply packages/client/base/src/types/hosted/v3/CrossmintHostedCheckoutV3Props.ts:4]: yeah its in the v3 doc 

theres really not much to review tho, its 95% the same as embed - only diff is appearance object, and payment doesnt take a `payer` signer object

image

https://docs.google.com/document/d/1y602C9MwV8HZi9aLTDR0KANOcvnTZEP0tvqmE_OqXDA/edit?tab=t.0

File: packages/client/ui/react-ui/src/components/hosted/v3/CrossmintHostedCheckoutV3.tsx [hunk 3]
```diff
@@ -0,0 +1,44 @@
+import { useCrossmint } from "@/hooks";
+import { createCrossmintApiClient } from "@/utils/createCrossmintApiClient";
+import { crossmintHostedCheckoutV3Service, type CrossmintHostedCheckoutV3Props } from "@crossmint/client-sdk-base";
+import { type MouseEvent, useState, type JSX } from "react";
+
+export type CrossmintHostedCheckoutV3ReactProps = CrossmintHostedCheckoutV3Props & JSX.IntrinsicElements["button"];
+
+export function CrossmintHostedCheckout_Alpha(props: CrossmintHostedCheckoutV3ReactProps) {
+    const [showOverlay, setShowOverlay] = useState(false);
```
Comment (pgarciaegido) [inline packages/client/ui/react-ui/src/components/hosted/v3/CrossmintHostedCheckoutV3.tsx:9]: remove

**Review by pgarciaegido — APPROVED** [review]

**Pull request merged**